ibm_cloud_sdk_core/utils.py

```diff
@@ -119,62 +128,62 @@ def read_from_credential_file(service_name, separator='='):
             for line in fp:
                 key_val = line.strip().split(separator)
                 if len(key_val) == 2:
-                    key = key_val[0].lower()
+                    key = key_val[0]
                     value = key_val[1]
                     _parse_key_and_update_config(config, service_name, key, value)
     return config
 
 def _parse_key_and_update_config(config, service_name, key, value):
-    if service_name in key:
-        index = key.find('_')
-        if index != -1:
-            config[key[index + 1:]] = value
+    service_name = service_name.replace(' ', '_').replace('-', '_').upper()
+    if key.startswith(service_name):
+        config[key[len(service_name) + 1:]] = value
 
 def read_from_vcap_services(service_name):
-    service_name = service_name.replace(' ', '_').lower()
     vcap_services = getenv('VCAP_SERVICES')
-    vcap_service_credentials = None
+    vcap_service_credentials = {}
     if vcap_services:
         services = json_import.loads(vcap_services)
 
         for key in services.keys():
-            name = key.replace('-', '_')
-            if name == service_name:
+            if key == service_name:
                 vcap_service_credentials = services[key][0]['credentials']
                 if vcap_service_credentials is not None and isinstance(vcap_service_credentials, dict):
                     if vcap_service_credentials.get('username') and vcap_service_credentials.get('password'): # cf
-                        vcap_service_credentials['auth_type'] = 'basic'
+                        vcap_service_credentials['AUTH_TYPE'] = 'basic'
+                        vcap_service_credentials['USERNAME'] = vcap_service_credentials.get('username')
+                        vcap_service_credentials['PASSWORD'] = vcap_service_credentials.get('password')
                     elif vcap_service_credentials.get('apikey'):  # rc
-                        vcap_service_credentials['auth_type'] = 'iam'
+                        vcap_service_credentials['AUTH_TYPE'] = 'iam'
+                        vcap_service_credentials['APIKEY'] = vcap_service_credentials.get('apikey')
                     else: # no other auth mechanism is supported
-                        vcap_service_credentials = None
+                        vcap_service_credentials = {}
     return vcap_service_credentials
 
-def contruct_authenticator(config):
-    auth_type = config.get('auth_type').lower() if config.get('auth_type') else 'iam'
+def _construct_authenticator(config):
+    auth_type = config.get('AUTH_TYPE').lower() if config.get('AUTH_TYPE') else 'iam'
```

Comment:
I didn't comment on this before, and I don't think we need to change anything right now, but... when we implement the service config feature, it might be good to define constants for the various config properties rather than use the literal strings like this.